resources/scponly_user.rb

```diff
@@ -0,0 +1,126 @@
+# true To learn more about Custom Resources, see https://docs.chef.io/custom_resources.html
+
+resource_name :scponly_user
+
+default_action :create
+
+property :write_dir, String, default: 'incoming'
+property :public_key, String
+property :private_key, String
+property :chroot, [true, false], default: true
+property :altroot, String
+property :binaries,
+          Array,
+          default:
+          %w(
+            /bin/chgrp
+            /bin/chmod
+            /bin/chown
+            /bin/ln
+            /bin/ls
+            /bin/mkdir
+            /bin/mv
+            /bin/rm
+            /bin/rmdir
+            /usr/bin/scp
+            /usr/sbin/scponlyc
+          )
+
+action :create do
+  run_context.include_recipe 'osl-scponly::default'
+
+  if new_resource.chroot
+
+    altroot = new_resource.altroot.nil? ? '/home/chroot' : new_resource.altroot
+
+    directory "#{altroot}/home/#{new_resource.name}" do
+      recursive true
+    end
+
+    user new_resource.name do
+      gid 'scponly'
```

Comment:
```suggestion
      gid new_resource.name
```

Reply:
Wouldn't that be default behavior of user add? should I just remove gid?

Reply:
That's because you need to create the group before you create the user. If you move that resource above the user resource, it should work properly.

Reply:
...oh my god. I am embarrassed. Also though I'm still curious shouldn't create user also create the user as a group and add the user to that group if gid is omitted?

Reply:
I believe ``adduser`` will do that but ``useradd`` won't by design.